### ISSUE TITLE
Clarify Hyperband docs/examples

### ISSUE THREAD

**Issue by samuela** [baggepinnen/Hyperopt.jl#34]
The Hyperband example in the readme looks exciting! I'm a little confused on how it lines up with the Hyperband algorithm presented elsewhere, e.g. in [Kevin's writeup](https://homes.cs.washington.edu/~jamieson/hyperband.html), however. 
```julia
ho = @hyperopt for i=18, sampler=Hyperband(R=50, η=3, inner=RandomSampler()), a = LinRange(1,5,1800), c = exp10.(LinRange(-1,3,1800))
    if state === nothing # Query if state is initialized
        res = optimize(resources=i, a, b) # if state is uninitialized, start a new optimization using the selected hyper parameters
    else
        res = optimize(resources=i, state=state) # If state has a value, continue the optimization from the state
    end
    minimum(res), get_state(res) # return the minimum value and a state from which to continue the optimization
end
```
In particular, `R` and `eta` appear to correspond with [the paper](https://arxiv.org/abs/1603.06560), but how does `i` fit into the Hyperband algorithm? What does `i=18` mean? Why are things run with `resources=i`? 

Thanks in advance!

**Comment by baggepinnen** [comment]
Hey!

`i=18` is simply the maximum number of iterations, the same as for the other samplers. The entire call to `optimize` in the example is made up, you simply call whatever function you want to call and assign it whatever amount of resources. In the example, I simply assign a hypothetical keyword argument `resources=i`, but you can make it as complicated as you want, e.g., 
```julia
r = calc_next_round_resources(state)
optimize_next_round(r, other state)
```

**Comment by samuela** [comment]
So it seems that `i` in the context of `@hyperopt for i=18` means something quite different than the `i` in the body of the expression. In the body of the expression, it seems like `i` corresponds to `r_i` in the paper. This is confusing because for example,
```julia
using Hyperopt

ho = @hyperopt for i in 18, sampler in Hyperband(R=50, η=3, inner=RandomSampler())
    @show i, state
    if state === nothing
        state = randn(5)
    end
    0, state
end
```
outputs among other things:
```
(i, state) = (50, nothing)
```
which is much larger than 18!

**Comment by samuela** [comment]
In fact,
```julia
ho = @hyperopt for i in -1, sampler in Hyperband(R=50, η=3, inner=RandomSampler())
```
does exactly the same thing as
```julia
ho = @hyperopt for i in 18, sampler in Hyperband(R=50, η=3, inner=RandomSampler())
```

**Comment by baggepinnen** [comment]
Hmm, I haven't actually used Hyperband myself since I implemented it so I might very well be wrong in how it works. It seems it decides the number of iterations by
```julia
iters = floor(Int,sampler.R*sampler.η^smax)
```
so the initial `i` might be unused for this sampler.

**Comment by baggepinnen** [comment]
I've realized that `Hyperband` behaves quite different compared to the other samplers, mostly because it did not really fit into the framework in which I implemented all other samplers :/ If you have any suggestions for how to improve the implementation I'd be happy to discuss. I added an example to the readme. All samplers are tested in the test file as well so if there is anything unclear, it might pay off to have a look there, it should be quite readable.
